src/azul/portal_service.py

```diff
@@ -196,7 +200,7 @@ def _read_db(self, version: str) -> JSONs:
             response = self.client.get_object(Bucket=self.bucket,
                                               Key=self.object_key,
                                               VersionId=version)
-        except self.client.exceptions.NoSuchKey:
+        except (self.client.exceptions.NoSuchKey, ClientError):
```

Comment:
Q for @hannes-ucsc:

This exception handling might not be surgical enough, ClientError casts a wide net of exceptions it will accept, and might hide an actual issue, should we cast wide and filter down? I think `NoSuchKey` would be good to except but that is not the correct exception for an `MissingVersionId`.

https://github.com/spulec/moto/issues/3884

Reply:
alternatively we could also remove the Azul class `NoSuchObjectVersion` and just raise the exception that was caught.

Reply:
explain hypothesis of eventual-constancy on line 199